### PR TITLE
Feature/property strategy config

### PR DESCRIPTION
**Pull request by no-reply**
@tpendragon will this work for you?

### CODE REVIEW

**Comment by tpendragon** [comment]
This looks reasonable. I'm not sure yet if it works for AF, but I can't give it a shot at the moment. :+1: to merging and patching later, if it turns out AF needs some other functionality.

**Comment by tpendragon** [comment]
@elrayle Any complaints?

**Comment by elrayle** [comment]
I'm going to look at this and PR #203 tomorrow.

**Comment by no-reply** [comment]
Thanks @elrayle. I'll try to be responsive here and on IRC about any issues with either of these PRs.

**Comment by elrayle** [comment]
This looks good to me.  If I don't here any other comments by 2:00 ET, I'll merge this.

**Comment by elrayle** [comment]
@mjsuhonos @no-reply Does this address issue #198?

**Comment by no-reply** [comment]
> Does this address issue #198?

Seems like. @mjsuhonos?

**Comment by mjsuhonos** [comment]
If I'm understanding correctly, this will set the default persistence strategy on a relation, but not the relating resource (ie. the subject) -- so this is a partial solution to #198.

I would propose extending this approach to allow a `:persist_to` argument to `RDFSource#initialize` to fully implement #198 as well as clean up [the logic mentioned in point 2](https://github.com/ActiveTriples/ActiveTriples/blob/e90d5e5d3dff1330913fb876e26981d6d811b48d/lib/active_triples/rdf_source.rb#L94-L99).

I can't get to this today but am happy to provide a PR if this seems reasonable to everyone.

**Comment by elrayle** [comment]
@no-reply @mjsuhonos Do we want to merge this as a partial fix?  Or wait for @mjsuhonos PR?

I just did a rebase, so when the tests complete, this will be ready to merge.

**Comment by no-reply** [comment]
:+1: to merging as a partial fix and leaving the ticket open

@mjsuhonos I'm :+1: on the proposed initializer change; I think that should be part of a larger rewrite for the initializer. I'm curious how you see these two ways of setting a persistence strategy interacting if/when they disagree.

**Comment by mjsuhonos** [comment]
> :+1: to merging as a partial fix and leaving the ticket open
Also :+1: here.

> I'm curious how you see these two ways of setting a persistence strategy interacting if/when they disagree.
Indeed; I may need some help with use cases / specs when the time comes.  Uncoupling the initializer logic and delegating to the respective modules (e.g. `Persistable`) will probably go a long way here.

File: spec/active_triples/relation_spec.rb [hunk 1]
```diff
@@ -478,16 +494,19 @@ class WithTitle
...
-      it 'sets mulitple values' do
-        values = [:moomin, :snork]
-        expect { subject.set(values) }
-          .to change { subject.to_a }.to contain_exactly(*values)
+        it 'returns values with persistence strategy set' do
+          expect(subject.set(RDF::Node.new).map(&:persistence_strategy))
+            .to contain_exactly(an_instance_of(ActiveTriples::RepositoryStrategy))
+        end
       end
     end
```
Comment (elrayle) [inline spec/active_triples/relation_spec.rb]: @no-reply Is is correct that these tests were removed?  This may be a bad merge of a rebase conflict.

**Comment by no-reply** [comment]
I re-rebased off my local branch of this and I think I have a clean version up now.

**Pull request merged**